partiql-ast/src/main/java/org/partiql/ast/DatetimeField.java

```diff
@@ -107,7 +102,7 @@ public static DatetimeField parse(@NotNull String value) {
             case "SECOND": return SECOND();
             case "TIMEZONE_HOUR": return TIMEZONE_HOUR();
             case "TIMEZONE_MINUTE": return TIMEZONE_MINUTE();
-            default: return UNKNOWN();
+            default: throw new IllegalArgumentException("No enum constant DatetimeField." + value);
```

Comment:
(self-review): behavior follows what Kotlin does for its `valueOf()` method, which throws an `IllegalArgumentException` if the string value does not match any of the enum variants.